### PR TITLE
#21697 adding the create type from base type

### LINKED ISSUE #21697: Copy ContentType API
**Is your feature request related to a problem? Please describe.**
There are circumstances that might require the creation of many content types that share the same base fields. Creating the same fields each time adds tedium and could result in user error.

**Describe the solution you'd like**
There are multiple potential ways to implement this, however after some discussion we think the best solution would be to allow a Content Type to copy its initial fields from another Content Type. This would allow users to create bootstrap Content Types for a lot of default fields, or extend existing already created Content Types.

### PR DESCRIPTION
**Pull request by jdotcms**
This PR is a draft to create a content type based on another content type (so you can reuse existing types to create a new ones)

### CODE REVIEW

**Review by erickgonzalez — CHANGES_REQUESTED** [review]
Here are some improvements from my POV:
* change the endpoint to `http://localhost:8080/api/v1/contenttype/{sourceContentTypeVarNameOrId}/_copy` to keep what we've been doing regarding names and functions.
* As a body use a JSON like the ContentTypeForm (the one we use to create/update content types) and for every property that is not sent in the JSON use the one from the **sourceContentType**

**Comment by jdotcms** [comment]
> Here are some improvements from my POV:
> 
> * change the endpoint to `http://localhost:8080/api/v1/contenttype/{sourceContentTypeVarNameOrId}/_copy` to keep what we've been doing regarding names and functions.
> * As a body use a JSON like the ContentTypeForm (the one we use to create/update content types) and for every property that is not sent in the JSON use the one from the **sourceContentType**

done

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/contenttype/CopyContentTypeForm.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import com.dotcms.repackage.javax.validation.constraints.NotNull;
+
+/**
+ * This form encapsulates the info for contentlet copy
+ * @author jsanca
+ */
+@JsonDeserialize(builder = CopyContentTypeForm.Builder.class)
+public class CopyContentTypeForm extends Validated {
+
+    @NotNull
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/contenttype/CopyContentTypeForm.java:16]: remove @NotNull

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/contenttype/CopyContentTypeForm.java (hunk 1 above)
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/contenttype/CopyContentTypeForm.java:16]: This is ok, is the only required property

**Review by erickgonzalez — CHANGES_REQUESTED** [review]

**Review by erickgonzalez — APPROVED** [review]

**Pull request merged**